letsencrypt/client.py

```diff
@@ -215,14 +215,17 @@ def obtain_certificate_from_csr(self, domains, csr,
 
         logger.debug("CSR: %s, domains: %s", csr, domains)
 
-        authzr = self.auth_handler.get_authorizations(domains)
+        if authzr is False:
+            authzr, _ = self.auth_handler.get_authorizations(
+                            domains,
+                            self.config.allow_subset_of_names)
```

Comment:
We shouldn't pass in `self.config.allow_subset_of_names` here. If we do and we are unable to obtain an authorization for a requested domain, `boulder` will give us an error because we don't have authorizations for all of the names in the CSR.